numpy/_core/code_generators/generate_umath.py

```diff
@@ -1142,6 +1142,14 @@ def english_upper(s):
           TD(O),
           signature='(n?,k),(k,m?)->(n?,m?)',
           ),
+'vecdot':
+    Ufunc(2, 1, None,
+          docstrings.get('numpy._core.umath.vecdot'),
+          "PyUFunc_SimpleUniformOperationTypeResolver",
+          TD(notimes_or_obj),
+          TD(O),
```

Comment:
Partially note to self: currently, `OBJECT` is incorrect, since it does not do the conjugate and the object might hold complex values. Probably easiest to just remove support for it.

Reply:
I added support for object (this crashes if non-numerical, so that there is no `conjugate()` method, but that seems OK)